src/tests/blueprint/t_blueprint_mpi_mesh_transform.cpp

```diff
@@ -0,0 +1,381 @@
+{
+    Node mesh, info;
+
+    int par_rank;
+    int par_size;
+    MPI_Comm_rank(MPI_COMM_WORLD, &par_rank);
+    MPI_Comm_size(MPI_COMM_WORLD, &par_size);
+
+    if (par_size == 1) return;
+
+    if (par_rank == 0)
+    {
+        Node& domain_0 = mesh["domain_000000"];
+
+        domain_0["state/domain_id"] = 0;
+        domain_0["state/level_id"] = 0;
+
+        std::vector<double> xcoords_0;
+        for (int i = 0; i < 5; ++i)
+        {
+            xcoords_0.push_back(0.0);
+            xcoords_0.push_back(0.25);
+            xcoords_0.push_back(0.5);
+            xcoords_0.push_back(0.75);
+            xcoords_0.push_back(1.0);
+        }
+        std::vector<double> ycoords_0;
+        double yval = 0.0;
+        for (int i = 0; i < 5; ++i)
+        {
+            ycoords_0.push_back(yval);
+            ycoords_0.push_back(yval);
+            ycoords_0.push_back(yval);
+            ycoords_0.push_back(yval);
+            ycoords_0.push_back(yval);
+            yval += 0.25;
+        }
+
+
+        Node& coords_0 = domain_0["coordsets/coords"];
+        coords_0["type"] = "explicit";
+        coords_0["values/x"].set(xcoords_0);
+        coords_0["values/y"].set(ycoords_0);
+
+        Node& topo_0 = domain_0["topologies/topo"];
+        topo_0["coordset"] = "coords";
+        topo_0["type"] = "structured";
+
+        topo_0["elements/origin/i0"] = 0;
+        topo_0["elements/origin/j0"] = 0;
+        topo_0["elements/dims/i"] = 4;
+        topo_0["elements/dims/j"] = 4;
+
+        Node& field_0 = domain_0["fields/field"];
+        field_0["association"] = "element";
+        field_0["type"] = "scalar";
+        field_0["topology"] = "topo";
+
+        std::vector<double> vals_0(16, 0.5);
+        field_0["values"].set(vals_0);
+
+        Node& adjset_0 = domain_0["adjsets/adjset"];
+        adjset_0["association"] =  "vertex";
+        adjset_0["topology"] =  "topo";
+
+        Node& group_0_1 = adjset_0["groups/group_000001"];
+        std::vector<int> nbrs_0(2);
+        nbrs_0[0] = 0;
+        nbrs_0[1] = 1;
+        group_0_1["neighbors"].set(nbrs_0);
+        group_0_1["rank"] = 1;
+
+        group_0_1["windows/window_000000/level_id"] = 0;
+        group_0_1["windows/window_000000/origin/i"] = 4;
+        group_0_1["windows/window_000000/origin/j"] = 0;
+        group_0_1["windows/window_000000/dims/i"] = 1;
+        group_0_1["windows/window_000000/dims/j"] = 5;
+        group_0_1["windows/window_000000/ratio/i"] = 2;
+        group_0_1["windows/window_000000/ratio/j"] = 2;
+        group_0_1["windows/window_000001/level_id"] = 1;
+        group_0_1["windows/window_000001/origin/i"] = 8;
+        group_0_1["windows/window_000001/origin/j"] = 0;
+        group_0_1["windows/window_000001/dims/i"] = 1;
+        group_0_1["windows/window_000001/dims/j"] = 9;
+        group_0_1["windows/window_000001/ratio/i"] = 2;
+        group_0_1["windows/window_000001/ratio/j"] = 2;
```

Comment:
It looks like this code was lifted from the previous example; would it be possible to abstract this into a helper function?

Reply:
Yes, this is true.